cdap-data-fabric/src/main/java/io/cdap/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -287,9 +281,9 @@ void create(String namespaceId, String name, DatasetInstanceConfiguration props)
     if (DatasetsUtil.isUserDataset(datasetId)) {
       LOG.trace("Authorizing impersonation for dataset {}", name);
       if (effectiveOwner != null) {
-        authorizationEnforcer.enforce(effectiveOwner, requestingUser, Action.ADMIN);
+        accessEnforcer.enforce(effectiveOwner, requestingUser, AccessPermission.SET_OWNER);
       }
-      authorizationEnforcer.enforce(datasetId, requestingUser, Action.ADMIN);
+      accessEnforcer.enforce(datasetId, requestingUser, StandardPermission.CREATE);
```

Comment:
I still want to pass a name to the access enforcer if it's provided. E.g. When namespace is created I want to pass a name in case enforcer would want to use it. Also I don't want to make namespace create an update operation (even though name is provided)

Reply:
Sounds good